cmd/boot-script-service/serviceAPI.go

```diff
@@ -78,24 +83,42 @@ func serviceStatusAPI(w http.ResponseWriter, req *http.Request) {
 			rsp.Body.Close()
 		}
 	}
-	if strings.Contains(strings.ToUpper(req.URL.Path), "ETCD") ||
+	if strings.Contains(strings.ToUpper(req.URL.Path), "STORAGE") &&
+		strings.Contains(strings.ToUpper(req.URL.Path), "STATUS") ||
 		strings.Contains(strings.ToUpper(req.URL.Path), "ALL") {
-		bssStatus.EctdStatus = "connected"
-		randnum := rand.Intn(255)
-		err := etcdTestStore(randnum)
-		if err != nil {
-			httpStatus = http.StatusInternalServerError
-			bssStatus.EctdStatus = "error"
-			log.Printf("Test store to etcd failed: %s", err)
+		var sb storageBackend
+		bssStatus.StorageBackend = &sb
+		if useSQL {
+			sb.Name = "postgres"
+			sb.Status = "connected"
+
+			if bp, err := bssdb.GetBootParamsAll(); err != nil {
+				httpStatus = http.StatusInternalServerError
+				log.Printf("Test access to postgres failed: %v", err)
+				sb.Status = "error"
+			} else {
+				log.Println("Test access to postgres using postgres.BootParamsGetAll() succeeded")
+				debugf("Boot parameters returned: %v", bp)
+			}
 		} else {
-			ret, err := etcdTestGet()
-			if err != nil || ret != randnum {
+			sb.Name = "etcd"
+			sb.Status = "connected"
+			randnum := rand.Intn(255)
+			err := etcdTestStore(randnum)
+			if err != nil {
 				httpStatus = http.StatusInternalServerError
-				bssStatus.EctdStatus = "error"
-				if err != nil {
-					log.Printf("Test read from etcd failed: %s", err)
-				} else {
-					log.Printf("Test read from etcd miscompare: Expected %d, Actual %d", randnum, ret)
+				sb.Status = "error"
+				log.Printf("Test store to etcd failed: %s", err)
```

Comment:
Should the `%s` here be either `%v` or `%w`? I don't know what `etcdTestStored()` is returning so I'm not sure.

Reply:
Fixed in adf2998.